interceptors/tracing2/interceptors.go

```diff
@@ -0,0 +1,70 @@
+package tracing2
```

Comment:
Why not modifying `tracing` directly?

Reply:
It makes the code easier to review.

I will rename it back to `tracing` in the last commit.